### ISSUE TITLE
Is there an alternative to using `<scan>` + jetty:start with Jetty 10+

### ISSUE THREAD

**Issue by Artur-** [eclipse/jetty.project#7451]
**Question**
We have a couple of integration tests in the Vaadin Flow project that test features related to live reload using Jetty. They are currently run with Jetty 9 using `jetty:start` + `jetty:stop` bound to `pre-integration-test` and `post-integration-test` and Jetty 9 has been configured using `<scanIntervalSeconds>1</scanIntervalSeconds>`.

The tests themselves modify the files in the `target` folder in some way and then wait for the modification to become visible in a browser.

Now it seems that Jetty 10+ no longer takes `<scan>` into account when using `jetty:start` so the tests no longer work after upgrading.

The tests themselves do not really depend on Jetty scanning and picking up changes but they do depend on a redeploy taking place. Is there any other "easy" way to cause Jetty started with `jetty:start` to redeploy?

**Comment by janbartel** [comment]
@Artur-  it seems like you have some integration tests that depend on an unintended consequence in jetty-9 that was eliminated in jetty-10 :)  Specifically, it wasn't intended that if you run `start` that a thread would also be started to scan for changes and redeploy, that was more of a consequence of many development iterations on the plugin that added features that resulted in some weird combination effects - most projects would not be expecting to run jetty on a pre-compiled project during the test phase and suddenly have jetty redeploy at some indeterminate points during that test phase.

Can you post a link to your tests, and describe exactly what you are trying to test and see if we can figure a way to achieve what you want?

**Comment by Artur-** [comment]
Thanks for confirming my suspicions and for asking what the tests actually do so I had to dig in and find out :) 

I think I found a handful of tests in total relying on this behavior and they test two different things:
1. Vaadin runs a webpack server in the background. This webpack server is re-used after redeploying the application during development. The test simulates this case and ensures that the same webpack instance really is used after a jetty redeploy. Then another test verifies some related implications of keeping the same webpack instance.
2. There are some special cases where the full http session is not serializable but some parts should still be serialized and retained while the unserializable parts are left out. This is tested by setting up the real scenario, creating a session, redeploying to Jetty and checking the session is still the way it is supposed to be.

https://github.com/vaadin/flow/tree/master/flow-tests/test-jetty-reload is a good example that seems to actually test both cases 1 and 2 above

**Comment by Artur-** [comment]
I think we might be able to work around this by using Spring Boot + `spring-boot-devtools` instead for the test. That combination has the reload behavior we are looking for

**Comment by janbartel** [comment]
@Artur- I'm not really sure what a `webpack` is, but it seems to be something to do with the static content of a webapp? In which case a jetty redeploy would never touch those anyway (unless you had assembled your webapp into a `war` that jetty would need to extract before deploying it). 

For the http session serialization, you don't actually need to test that by stopping and restarting jetty. You could test that by configuring one of the passivating session data stores (eg FileSessionDataStore) and ensuring that your listeners that do the pre-passivation clear out of non-serializable attributes are called. But also, please rest assured that we have unit tests in place for testing the calling of these listeners in jetty too :)

BTW I get a 404 on that link you posted?

**Comment by Artur-** [comment]
Thanks for the help.

The tests try to test what happens in the real world in redeployment situations so mocking or doing custom configuration is not wanted for these particular cases. And for sure, they were not testing Jetty logic but our logic :)

Seems like the test was moved after I posted the link so it now gives 404. It is now at https://github.com/vaadin/flow/tree/master/flow-tests/test-redeployment

We ended up changing the problematic tests to use Spring Boot and spring boot devtools, which do what we need.
